### PR TITLE
Add initial 1.4 upgrade notice

### PR DESCRIPTION
**Pull request by howardjohn**
I think it makes sense to check in a first iteration of this, then as
individuals add their own sections we can merge those in iteratively.

TODO: if this merges open up an issue to upgrade the link here to 1.4 release notes -- I have it linking to 1.3 release notes to pass lint

### CODE REVIEW

**Comment by howardjohn** [comment]
Fixes https://github.com/istio/istio.io/issues/5155

**Comment by geeknoid** [comment]
Mind if we close this PR in favor of https://github.com/istio/istio.io/pull/5138? My PR reorgs how we do release notes and upgrade notes, and includes placeholders for 1.4 already. My PR is ready to merge, just waiting for Frank to click the green button.

**Comment by howardjohn** [comment]
Sounds good. Once that merged I'll add the content on top of your changes

File: content/en/docs/setup/upgrade/notice/index.md [hunk 1]
```diff
@@ -1,65 +1,28 @@
...
+title: 1.4 Upgrade Notice
+description: Important changes to consider when upgrading to Istio 1.4.
 weight: 5
 aliases:
     - /docs/setup/kubernetes/upgrade/notice/
 ---
 
 This page describes changes you need to be aware of when upgrading from
-Istio 1.2 to 1.3.  Here, we detail cases where we intentionally broke backwards
+Istio 1.4 to 1.4.  Here, we detail cases where we intentionally broke backwards
```
Comment (ericvn) [inline content/en/docs/setup/upgrade/notice/index.md]: Should be 1.3 to 1.4

File: content/en/docs/setup/upgrade/notice/index.md [hunk 2]
```diff
@@ -1,65 +1,28 @@
...
 compatibility.  We also mention cases where backwards compatibility was
 preserved but new behavior was introduced that would be surprising to someone
-familiar with the use and operation of Istio 1.2.
+familiar with the use and operation of Istio 1.3.
 
-For an overview of new features introduced with Istio 1.3, please refer
-to the [1.3 release notes](/news/2019/announcing-1.3/).
+For an overview of new features introduced with Istio 1.4, please refer
+to the [1.4 release notes](/news/2019/announcing-1.3/).
```
Comment (ericvn) [inline content/en/docs/setup/upgrade/notice/index.md]: announcing-1.4

**Comment by howardjohn** [comment]
@geeknoid thanks for the changes, I updated my pr to accommodate them

**Pull request merged**